pyomo/core/kernel/block.py

```diff
@@ -76,6 +76,21 @@ class block(IBlock):
 
     _lbs_count = 4
     _ctype = IBlock
+    _block_reserved_words = set()
+
+    @staticmethod
+    def _refresh_block_reserved_words():
+        block._block_reserved_words = set(dir(block()))
+        # these private attributes are frequently modified
+        block._block_reserved_words.remove('_parent')
+        block._block_reserved_words.remove('_storage_key')
+        block._block_reserved_words.remove('_active')
+        block._block_reserved_words.remove('_byctype')
+        block._block_reserved_words.remove('_order')
```

Comment:
By removing these from `_block_reserved_words`, you are removing any protections for these (important) attributes.  Wouldn't us be better to leave these in the protected list and just use `object.__setattr__(x)` where you use it?  Or if they really do get used that often, make a helper method that wraps the call to `object.__setattr__()`?